### PR TITLE
feat: add `shopify-cli` spec

### PR DESCRIPTION
**Pull request by beynar**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
**Feature** : this PR introduces autocompletion for the [shopify-cli](https://shopify.dev/apps/tools/cli/core-commands).

CleanShot 2022-03-19 at 19 07 35@2x

### CODE REVIEW

**Review by fedeci — CHANGES_REQUESTED** [review]
Thank you for this contribution! There are some things to fix but it looks really nice!

File: src/shopify.ts [hunk 1]
```diff
@@ -0,0 +1,818 @@
+const i = (name: string) => `fig://icon?type=${name}`;
+
+const helpOption: Fig.Option = {
+  name: ["--help", "-h"],
+  icon: i("option"),
+  description: "Show help for this command",
+  priority: 1,
+};
```
[the remark this answers is not in the document]
Comment (beynar) [reply src/shopify.ts:8]: nice !

File: src/shopify.ts [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+const configOptions: Fig.Option[] = [
+  {
+    name: "--status",
+  },
+  {
+    name: "--enable",
+  },
+  {
+    name: "--disable",
+  },
```
Comment (fedeci) [inline src/shopify.ts:25]: Can you also add a small description to these?

File: src/shopify.ts [hunk 3]
```diff
@@ -0,0 +1,818 @@
...
+                isOptional: false,
+              },
+            },
+            {
+              name: "--language",
+              description:
+                "He format or language of the script. Valid values: wasm. If unspecified, then your script project is created in WebAssembly (Wasm), which is the default format for script projects",
+              args: {
+                name: "script_language",
+                isOptional: false,
```
[the remark this answers is not in the document]
Comment (fedeci) [reply src/shopify.ts]: And here :)

File: src/shopify.ts [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+
+const appCreationBaseOptions: Fig.Option[] = [
+  helpOption,
+  {
+    name: "--name",
+    description: "App name. Any string",
+    args: {
+      name: "NAME",
+    },
+    isRequired: true,
```
[the remark this answers is not in the document]
Comment (fedeci) [reply src/shopify.ts]: Friendly ping here😄

File: src/shopify.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fedeci) [reply src/shopify.ts]: And here :)

File: src/shopify.ts (hunk 2 above)
Comment (mschrage) [reply src/shopify.ts:25]: If this is the only thing we are waiting on, then I think it's okay to merge this in cc @fedeci @beynar

**Review by fedeci — APPROVED** [review]

**Comment by fedeci** [comment]
Thanks!

**Pull request merged**